src/adapter/graphql/tests/tests/test_from_data_schema.rs

```diff
@@ -0,0 +1,196 @@
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use datafusion::arrow::array::{Int32Array, StringArray};
+use datafusion::arrow::datatypes::{DataType, Field, Schema};
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::common::DataFusionError;
+use datafusion::error::Result;
+use datafusion::prelude::*;
+use kamu_adapter_graphql::scalars::data_schema::{DataSchema, DataSchemaFormat};
+use serde_json::{to_string_pretty, Value};
+
+#[test_log::test(tokio::test)]
+async fn test_from_parquet_schema_parquet() -> Result<()> {
+    let schema = Arc::new(Schema::new(vec![
+        Field::new("a", DataType::Utf8, false),
+        Field::new("b", DataType::Int32, false),
+    ]));
+
+    let batch = RecordBatch::try_new(
+        schema,
+        vec![
+            Arc::new(StringArray::from(vec!["a", "b", "c", "d"])),
+            Arc::new(Int32Array::from(vec![1, 10, 10, 100])),
+        ],
+    )?;
+
+    let ctx = SessionContext::new();
+
+    ctx.register_batch("t", batch)?;
+    let df = ctx.table("t").await?;
+
+    let filter = col("b").eq(lit(10));
+
+    let df = df.select_columns(&["a", "b"])?.filter(filter)?;
+
+    let result = DataSchema::from_data_frame_schema(df.schema(), DataSchemaFormat::Parquet);
+
+    assert!(result.is_ok());
+    let data_schema = result.unwrap();
+
+    let expected_content = indoc::indoc!(
+        r#"
+    message arrow_schema {
+      REQUIRED BYTE_ARRAY a (STRING);
+      REQUIRED INT32 b;
+    }
+    "#
+    );
+
+    let actual_content = data_schema.content.as_str();
+
+    assert_eq!(data_schema.format, DataSchemaFormat::Parquet);
+    assert_eq!(actual_content, expected_content);
+
+    Ok(())
+}
+
+#[test_log::test(tokio::test)]
+async fn test_from_parquet_schema_parquet_json() -> Result<()> {
+    let schema = Arc::new(Schema::new(vec![
+        Field::new("a", DataType::Utf8, false),
+        Field::new("b", DataType::Int32, false),
+    ]));
+
+    let batch = RecordBatch::try_new(
+        schema,
+        vec![
+            Arc::new(StringArray::from(vec!["a", "b", "c", "d"])),
+            Arc::new(Int32Array::from(vec![1, 10, 10, 100])),
+        ],
+    )?;
+
+    let ctx = SessionContext::new();
+
+    ctx.register_batch("t", batch)?;
+    let df = ctx.table("t").await?;
+
+    let filter = col("b").eq(lit(10));
+
+    let df = df.select_columns(&["a", "b"])?.filter(filter)?;
+
+    let result = DataSchema::from_data_frame_schema(df.schema(), DataSchemaFormat::ParquetJson);
+
+    assert!(result.is_ok());
+    let data_schema = result.unwrap();
+
+    let expected_content = indoc::indoc!(
+        r#"{
+        "fields": [
+          {
+            "logicalType": "STRING",
+            "name": "a",
+            "repetition": "REQUIRED",
```

Comment:
I'm concerned that this test will start failing if any of these fields get reordered in arrow code or by json serializer. I think we should parse `data_schema.content` into `serde_json::Value` and then compare against a value constructed via `json!({ .. })` macro. This way it will be resistant to formatting changes.

Reply:
Same for the `arrow-json` test below

Reply:
changed